Physics/src/pointmass.cpp

```diff
@@ -49,6 +29,111 @@ void PointMass::integrate(real duration) {
 	clearAccumulator();
 }
 
+real PointMass::getKineticEnergy() {
+	return 0.5 * (1.0f / inverseMass) * velocity.magnitude();
+}
+
+// ------------ Mass ------------
+
+real PointMass::getMass() const
+{
+	if (PointMass::inverseMass <= 0.0)
+	{
+		return REAL_MAX;
+	}
+	else
+	{
+		return ((real)1.0) / inverseMass;
+	}
+}
+
+void PointMass::setMass(real mass) {
+	inverseMass = (real)1.0 / mass;
+}
+
+real PointMass::getInverseMass() const
+{
+	return inverseMass;
+}
+
+void PointMass::setInverseMass(real invMass) {
+	inverseMass = invMass;
+}
+
+// ------------ Position ------------
+
+void PointMass::getPosition(Vector3* position) const
+{
+	*position = PointMass::position;
```

Comment:
If the purpose of this function is to assign some other object, we should rename the function

Reply:
Typically if you want to make it very clear that the pointer is an "out" variable we prefix the variable name with OUT or out 

i.e. outPosition

Reply:
Calling other physics contributors to weigh in on naming of the void getters @imnorookie @jchouhan19

Reply:
I think keeping it as a getter is fine but renaming the formal parameter to "out" for clarity reasons is a good idea thaWanks foWor that idea RichaWrd :333. 

This accessor is like this for performance reasons, so it isn't returning a new Vector3 object every time we're checking and applying positional updates. This is because return happens by value in CPP, meaning it can be very costly if we are simulating a ton of physics bodies in a single PhysicsWorld time-step. We'd be creating a new Vector3 object every single time.

My compromise would be to rename the formal parameter to "out" and then also make a second getter that returns a Vector3 copy as well, should someone interfacing with our Physics code want it. BeWest of BoWoth woWorlds! Fix it nao!!!!